bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/handler/BondDeviceHandler.java

```diff
@@ -0,0 +1,822 @@
+            updateState(CHANNEL_UP_LIGHT_DIRECTIONAL_INC, OnOffType.OFF);
+            updateState(CHANNEL_UP_LIGHT_DIRECTIONAL_DECR, OnOffType.OFF);
+            updateState(CHANNEL_DOWN_LIGHT_START_STOP, OnOffType.OFF);
+            updateState(CHANNEL_DOWN_LIGHT_DIRECTIONAL_INC, OnOffType.OFF);
+            updateState(CHANNEL_DOWN_LIGHT_DIRECTIONAL_DECR, OnOffType.OFF);
+            updateState(CHANNEL_STOP, OnOffType.OFF);
+            updateState(CHANNEL_HOLD, OnOffType.OFF);
+
+        } else {
+            logger.debug("No state information provided to update channels with");
+        }
+    }
+
+    private boolean hasConfigurationError() {
+        ThingStatusInfo statusInfo = getThing().getStatusInfo();
+        return statusInfo.getStatus() == ThingStatus.OFFLINE
+                && statusInfo.getStatusDetail() == ThingStatusDetail.CONFIGURATION_ERROR;
+    }
+
+    private synchronized boolean wasBindingUpdated() {
+        // Check if the binding has been updated
+        boolean updatedBinding = true;
+        @Nullable
+        String lastBindingVersion = this.getThing().getProperties().get(PROPERTIES_BINDING_VERSION);
+        updatedBinding = !CURRENT_BINDING_VERSION.equals(lastBindingVersion);
+        if (updatedBinding) {
+            logger.info("Bond Home binding has been updated.");
+            logger.info("Current version is {}, prior version was {}.", CURRENT_BINDING_VERSION, lastBindingVersion);
+
+            // Update the thing with the new property value
+            final Map<String, String> newProperties = new HashMap<>(thing.getProperties());
+            newProperties.put(PROPERTIES_BINDING_VERSION, CURRENT_BINDING_VERSION);
+
+            final ThingBuilder thingBuilder = editThing();
+            thingBuilder.withProperties(newProperties);
+            updateThing(thingBuilder.build());
+        }
+        return updatedBinding;
+    }
+
+    private synchronized boolean wasThingUpdatedExternally(BondDevice devInfo) {
+        // Check if the Bond hash tree has changed
+        config = getConfigAs(BondDeviceConfiguration.class);
+        final String lastDeviceConfigurationHash = config.lastDeviceConfigurationHash;
+        boolean updatedHashTree = !devInfo.hash.equals(lastDeviceConfigurationHash);
+        if (updatedHashTree) {
+            logger.info("Hash tree of device has been updated by Bond.");
+            logger.info("Current state is {}, prior state was {}.", devInfo.hash, lastDeviceConfigurationHash);
+        }
+        return updatedHashTree;
+    }
+
+    private boolean getBridgeAndAPI() {
+        Bridge myBridge = this.getBridge();
+        if (myBridge == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "No Bond bridge is associated with this Bond device");
+            logger.error("No Bond bridge is associated with this Bond device - cannot create device!");
```

Comment:
See above. The log message could be removed. Same for below.

Reply:
done